content/articles/using-logistic-regression-in-pytorch-to-identify-handwritten-digits/index.md

```diff
@@ -0,0 +1,169 @@
+```Python
+class LogisticRegression(tn.Module):
+	def __init__(self, input_size, num_classes):
+		super(LogisticRegression, self).__init__()
+		self.linear = tn.Linear(input_size, num_classes)
+
+	def forward(self, x):
+		out = self.linear(x)
+		return out
+```
+Now that our class has been established, we may create an instance of it.
+```Python
+model = LogisticRegression(input_size, num_classes)
+```
+### Building The Neural Network
+Our loss function and optimizer are now set. As specified in the hyperparameter above, we'll use the cross-entropy loss and the stochastic gradient descent algorithm with a reading rate of 0.001 for the optimizer.
+Let's import some modules first before building a new class for the network we're developing.
+```python
+class NeuralNetwork(tn.Module):
+   def __init__(self):
+      super(NeuralNetwork, self).__init__()
+      self.c1 = tn.Conv2d(1, 10, kernel_size=5)
+      self.c2 = tn.Conv2d(10, 20, kernel_size=5)
+      self.c2_drop = tn.Dropout2d()
+      self.fch1 = tn.Linear(320, 50)
+      self.fch2 = tn.Linear(50, 10)
+   def forward(self, b):
+      b = tF.relu(tF.max_pool2d(self.c1(b), 2))
+      b = tF.relu(tF.max_pool2d(self.c2_drop(self.c2(b)), 2))
+      b = b.view(-1, 320)
+      b = tF.relu(self.fch1(b))
+      b = tF.dropout(b, trai=self.training)
+      b = self.fch2(b)
+      return tF.log_softmax(b)
+```
+To get the network and the optimizer up and running;
+```python
+criterion = tn.CrossEntropyLoss()
+network = NeuralNetwork()
+opti = torch.optim.SGD(newmodel.parameters(), lr = reading_rate)
+```
+We're ready to begin training now. Resetting all gradients to 0 will be the first step here, followed by a forward pass, the loss calculation, backpropagation, and the updating of all weights.
+```python
+# Training the Model
+for epoch in range(num_epochs):
+	for i, (images, labels) in enumerate(train_loader):
+		images = Variable(images.view(-1, 28 * 28))
+		labels = Variable(labels)
+
+		optimizer.zero_grad()
+		outputs = model(images)
+		loss = criterion(outputs, labels)
+		loss.backward()
+		optimizer.step()
+
+		if (i + 1) % 100 == 0:
+			print('Epoch: [% d/% d], Step: [% d/% d], Loss: %.4f'% (epoch + 1, num_epochs, i + 1, len(train_dataset) // batch_size, loss.data))
+```
+Finally, we'll run the model through its paces using the code below.
+```Python
+correct = 0
+total = 0
+for images, labels in test_loader:
+	images = Variable(images.view(-1, 28 * 28))
+	outputs = model(images)
+	_, predicted = torch.max(outputs.data, 1)
+	total += labels.size(0)
+	correct += (predicted == labels).sum()
+
+print('Accuracy of the model on the 10000 test images: % d %%' % (
+			100 * correct / total))
+
+```
+To put it another way, if you followed the instructions exactly, you would have an accuracy rate of 91%, which is significantly lower than the current best model, which makes use of a different sort of neural network architecture.

```

Comment:
This is an incorrect way to put it. Accuracy is a measure of the data and data-preprocessing model capacity. So instead of giving them a number they should attain, mention that by following the steps, you got 91%.